implementations/rust/ockam/ockam_api/tests/authority.rs

```diff
@@ -95,7 +95,7 @@ async fn one_admin_test_api(ctx: &mut Context) -> Result<()> {
     assert_eq!(members.len(), 1);
     let attrs = members.get(&admin.identifier).unwrap();
 
-    assert!(attrs.added() - now < 5.into());
+    assert!(attrs.added().abs_diff(now) < 5.into());
```

Comment:
Shouldn't we still try to test that `attrs.added() >= now`?

Reply:
We want to test that the value of `added` field is correct. There are several ways of defining "correct" here. I think the definition "lies inside some small interval around `now`" is absolutely sane here. I don't think testing `added >= now` is useful. I was also bitten recently by my wrong assumptions, which caused a bug, and I needed to add some drift allowance [here](https://github.com/build-trust/ockam/blob/75727da16d6f45b6ef81f3b4cc966db900365bba/implementations/rust/ockam/ockam_identity/src/purpose_keys/purpose_keys_verification.rs#L95). It should not be a problem when everything is on the same machine, but makes me think that it's an unnecessary check.

Reply:
It can indeed make sense to add an assertion to check if it doesn't take too long to add an attribute, as a sanity check that we don't inadvertently create a timestamp in 100 years for example. 

What annoys me with using an absolute diff is that it kind of puts causality under the carpet. I expect the attribute to be `added` _after_ the user called the function to add a member. If we end up with `added()` being consistently 3 seconds before the call to `add_member` this could make up some very confusing logs.

Reply:
```
What annoys me with using an absolute diff is that it kind of puts causality under the carpet. I expect the attribute to be added after the user called the function to add a member. If we end up with added() being consistently 3 seconds before the call to add_member this could make up some very confusing logs.
```
1. Casualty is not expected in this test but is totally fine.
2. While it's possible to set an incorrect Timestamp or mangle it during some conversions (e.g. confuse seconds and minutes), it's very hard to imagine an implementation that would be consistently off by -3 seconds.
3. We're spending too much time discussing an insignificant problem IMHO.

Reply:
> We're spending too much time discussing an insignificant problem IMHO

This is clearly not a major problem :-D. I am having some compassion for the poor fellow, like me, who will find this code peculiar in the future and will wonder why things were written that way. But I can see that our brains are wired differently :-)